src/aws-cpp-sdk-core/source/http/curl/CurlHandleContainer.cpp

```diff
@@ -44,7 +45,20 @@ CURL* CurlHandleContainer::AcquireCurlHandle()
         CheckAndGrowPool();
     }
 
-    CURL* handle = m_handleContainer.TryAcquire();
+    // TODO: 1.12: start to fail instead of infinite loop
+    CURL* handle = nullptr;
+    bool errorLogged = false;  // avoid log explosion on legacy app behavior
+    while (!handle && m_poolSize) {
+      constexpr unsigned long DEFAULT_TIMEOUT = 1000l;
+      handle = m_handleContainer.TryAcquire(std::max(m_connectTimeout, DEFAULT_TIMEOUT));
```

Comment:
i think we should use something else besides m_connectTimeout  as that maps one to one with `CURLOPT_CONNECTTIMEOUT_MS`, and we are sort of tying those together. this has its own unique meaning of "aquisition from the pool time", not "DNS lookup time". maybe create a static const/maybe add a config var if this would touch several clients.

Reply:
Removed usage of m_connectTimeout, and just set 1s of wait time.